server/src/main/java/io/crate/types/IpType.java

```diff
@@ -188,7 +189,11 @@ private void validate(String ip) {
 
     @Override
     public int compare(String val1, String val2) {
-        return val1.compareTo(val2);
+        // Convert the IPs to BigInteger, to have a consistent way to compare them, and
+        // also handle the cases of IPv4 comparing to an IPv6
+        BigInteger ip1 = new BigInteger(1, InetAddresses.ipStringToBytes(val1));
+        BigInteger ip2 = new BigInteger(1, InetAddresses.ipStringToBytes(val2));
+        return ip1.compareTo(ip2);
```

Comment:
No need to go through `BigInteger`:

```suggestion
        byte[] ip1 = InetAddresses.ipStringToBytes(val1);
        byte[] ip2 = InetAddresses.ipStringToBytes(val2);
        return Arrays.compare(ip1, ip2);
```

See also the `equals` implementation of `Inet4Address` and `Inet6Address`

Reply:
great! thx. Although we need `compareUnsigned` to make this work. See for example these 2 IPs:
```
val1 = "10.200.1.100"
val2 = "10.0.0.1"
ip1 = {byte[4]@16096} [10, -56, 1, 100]
ip2 = {byte[4]@16101} [10, 0, 0, 1]
```
Normal compare returns -56, denoting that ip1 < ip2 which is not correct.